File: py/1694.reformat-phone-number.py
#
# @lc app=leetcode id=1694 lang=python3
#
# [1694] Reformat Phone Number
#

# @lc code=start
class Solution:
    def reformatNumber(self, number: str) -> str:
        res = []
        count = 0
        for c in number:
            if c.isdigit():
                if count == 3:
                    res.append('-')
                    count = 0
                count += 1
                res.append(c)
        if len(res) > 1 and count == 1:
            res[-2], res[-3] = res[-3], res[-2]
        return ''.join(res)

# @lc code=end

File: cpp/1694.reformat-phone-number.cpp
/*
 * @lc app=leetcode id=1694 lang=cpp
 *
 * [1694] Reformat Phone Number
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string reformatNumber(string number)
    {
        string res = "";
        int count = 0;
        for (auto c : number)
        {
            if (isdigit(c))
            {
                ++count;
                res.push_back(c);
                if (count == 3)
                {
                    res.push_back('-');
                    count = 0;
                }
            }
        }
        if (res.size() > 1 && count == 1)
        {
            swap(res[res.size() - 2], res[res.size() - 3]);
        }
        else if (count == 0)
        {
            res.pop_back();
        }
        return res;
    }
};
// @lc code=end
